src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,191 @@
+            Node node = new Node(null, value, head);
+            head.prev = node;
+            node.next = head;
+            head = node;
+            size++;
+        } else {
+            if (index == size || size == 0) {
+                add(value);
+            } else {
+                Node node = new Node(head, value, head.next);
+                for (int i = 1; i < index; i++) {
+                    node.prev = node.next;
+                    node.next = node.next.next;
+                }
+                node.next.prev = node;
+                node.prev.next = node;
+                size++;
+            }
+        }
+
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validIndexCheck(index);
+        Node<T> node = head;
+        for (int i = 0; i < index; i++) {
+            node = node.next;
+        }
+        return node.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        validIndexCheck(index);
+        T oldValue;
+        if (index == 0) {
+            oldValue = head.item;
+            head.item = value;
+            return oldValue;
+        }
+        if (index == size - 1) {
+            oldValue = tail.item;
+            tail.item = value;
+        } else {
+            Node node = new Node(head, value, head.next);
+            for (int i = 1; i < index; i++) {
+                node.prev = node.next;
+                node.next = node.next.next;
+            }
+            oldValue = (T) node.next.item;
+            node.next.item = value;
+        }
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validIndexCheck(index);
+        Node<T> node = new Node<>(null,head.item, head.next);
+        T removedValue;
+        if (size == 1) {
+            removedValue = head.item;
+            head.item = null;
+            tail.item = null;
+        } else {
+            if (index == 0) {
+                removedValue = head.item;
+                head = head.next;
+                head.prev = null;
+                size--;
+                return removedValue;
+            }
+            if (index == size - 1) {
+                removedValue = tail.item;
+                tail = tail.prev;
+                tail.next = null;
+            } else {
+                for (int i = 0; i < index; i++) {
+                    node.prev = node.next;
+                    node.next = node.next.next;
+
+                }
+                node.item = node.prev.item;
+                removedValue = node.item;
+                node.item = null;
+                node.next.prev = node.prev.prev;
+                node.prev.prev.next = node.next;
+                node.prev = null;
+                node.next = null;
+            }
```

Comment:
The logic in the remove method by index is incorrect. You don't need to create a new node at the beginning. Instead, find the node that needs to be removed and then use the unlink method to remove it.